python/cudf/cudf/core/dataframe.py

```diff
@@ -3139,6 +3139,27 @@ def label_encoding(
 
         return outdf
 
+    @copy_docstring(Frame._sample)
+    def sample(
+        self,
+        n=None,
+        frac=None,
+        replace=False,
+        weights=None,
+        random_state=None,
+        axis=None,
+        keep_index=True,
+    ):
+        return self._sample(
+            n=n,
+            frac=frac,
+            replace=replace,
+            weights=weights,
+            random_state=random_state,
+            axis=axis,
+            keep_index=keep_index,
+        )
```

Comment:
Instead of doing this could we define `sample` at the `Frame` level and inherit it in DataFrame / Series?

Reply:
Didn't want to expose it in Index, so tried it this way.

Reply:
I think it's fine to expose it in Index. Other than Pandas not having it is there any reason not to?

Reply:
keep_index option would be little odd, but this was added just for cudf, doesn't exist in pandas.